sunpy/map/mapbase.py

```diff
@@ -1906,20 +1910,15 @@ def superpixel(self, dimensions: u.pixel, offset: u.pixel = (0, 0)*u.pixel, func
         new_ny = new_array.shape[0]
 
         # Update metadata
-        new_meta['cdelt1'] = (dimensions[0] * self.scale[0]).value
-        new_meta['cdelt2'] = (dimensions[1] * self.scale[1]).value
+        new_meta['cdelt1'] = scale_factor[0] * self.scale[0].to_value(u.deg / u.pix)
+        new_meta['cdelt2'] = scale_factor[1] * self.scale[1].to_value(u.deg / u.pix)
         if 'CD1_1' in new_meta:
-            new_meta['CD1_1'] *= dimensions[0].value
-            new_meta['CD2_1'] *= dimensions[0].value
-            new_meta['CD1_2'] *= dimensions[1].value
-            new_meta['CD2_2'] *= dimensions[1].value
-        new_meta['crpix1'] = (new_nx + 1) / 2.
-        new_meta['crpix2'] = (new_ny + 1) / 2.
-        lon, lat = self._get_lon_lat(self.center.frame)
-        new_meta['crval1'] = lon.to(self.spatial_units[0]).value + 0.5 * \
-            (offset[0]*self.scale[0]).to(self.spatial_units[0]).value
-        new_meta['crval2'] = lat.to(self.spatial_units[1]).value + 0.5 * \
-            (offset[1]*self.scale[1]).to(self.spatial_units[1]).value
+            new_meta['CD1_1'] *= scale_factor[0]
+            new_meta['CD2_1'] *= scale_factor[0]
+            new_meta['CD1_2'] *= scale_factor[1]
+            new_meta['CD2_2'] *= scale_factor[1]
+        new_meta['crpix1'] = ((self.meta['crpix1'] - 0.5) / scale_factor[0]) + 0.5 + offset[0]
+        new_meta['crpix2'] = ((self.meta['crpix2'] - 0.5) / scale_factor[1]) + 0.5 + offset[1]
```

Comment:
This doesn't look to be the correct handling of `offset`...  `offset` is in original-map pixels, but you shift `CRPIX` as if it is in  resampled-map pixels

Reply:
Thanks; I think it should be fixed now.